src/store/transactions/deposit/deposit.thunks.ts

```diff
@@ -68,33 +66,31 @@ function fetchAccounts(fiatExchangeRates: FiatExchangeRates, preferredCurrency:
     dispatch(depositActions.loadEthereumAccounts());
 
     if (wallet !== undefined) {
-      return CoordinatorAPI.getTokens(undefined, undefined, undefined, undefined, 2049).then(
-        (res) => {
-          ethereum
-            .getTokens(wallet, res.tokens)
-            .then((ethereumAccounts): EthereumAccount[] => {
-              return ethereumAccounts.map((ethereumAccount) => {
-                const tokenFromTokensPrice =
-                  tokensPriceTask.status === "successful" &&
-                  tokensPriceTask.data[ethereumAccount.token.id];
-                const token = tokenFromTokensPrice ? tokenFromTokensPrice : ethereumAccount.token;
-
-                const fiatBalance = convertTokenAmountToFiat(
-                  ethereumAccount.balance,
-                  token,
-                  preferredCurrency,
-                  fiatExchangeRates
-                );
-
-                return { ...ethereumAccount, balance: ethereumAccount.balance, fiatBalance };
-              });
-            })
-            .then((metaMaskTokens) =>
-              dispatch(depositActions.loadEthereumAccountsSuccess(metaMaskTokens))
-            )
-            .catch((err) => dispatch(depositActions.loadEthereumAccountsFailure(err)));
-        }
-      );
+      return persistence.getTokens(undefined, undefined, undefined, undefined, 2049).then((res) => {
+        ethereum
+          .getTokens(wallet, res.tokens)
+          .then((ethereumAccounts): EthereumAccount[] => {
+            return ethereumAccounts.map((ethereumAccount) => {
```

Comment:
I think that now would make sense to move this this logic to the `ethereum` persistence layer right?

Reply:
Done. I think it makes total sense. The change was not straight forward though. Please carefully check the commit 0f312c166c77fa0d5ed7bf77abdf0843c1a89306 since I had to:

* Add the params `fiatExchangeRates` and `preferredCurrency` to some functions.
* Make several changes in `getEthereumAccounts` implementation.

Let's make sure I did not break anything... 😓

Reply:
> 

By the way, do you think that `src/utils/ethereum.ts` belongs to persistence? I concluded it does not, but since you refer to it as "the `ethereum` persistence layer" we better discuss it.

Reply:
Ethereum util moved to adapters as agreed.